control-plane/api-gateway/controllers/index.go

```diff
@@ -325,3 +331,15 @@ func filtersForHTTPRoute(o client.Object) []string {
 	}
 	return filters
 }
+
+func gatewayForGatewayPolicy(o client.Object) []string {
+	gatewayPolicy := o.(*v1alpha1.GatewayPolicy)
+
+	targetGateway := gatewayPolicy.Spec.TargetRef
+	//gateway policy is 1to1
+	if targetGateway.Group == v1alpha1.ConsulHashicorpGroup && targetGateway.Kind == common.KindGateway {
+		return []string{common.IndexedNamespacedNameWithDefault(targetGateway.Name, &targetGateway.Namespace, gatewayPolicy.Namespace).String()}
+	}
```

Comment:
looking at this more, I don't think this works as expect because the `IndexedNamespacedNameWithDefault` expects the second argument to be nil-able but `targetGateway.Namespace` will never be nil so it will always use that value no matter what is in it (even an empty string), I think something like the following would work
```suggestion
func gatewayForGatewayPolicy(o client.Object) []string {
	gatewayPolicy := o.(*v1alpha1.GatewayPolicy)

	targetGateway := gatewayPolicy.Spec.TargetRef
	// gateway policy is 1to1
	if targetGateway.Group == "gateway.networking.k8s.io/v1beta1" && targetGateway.Kind == common.KindGateway {
		policyNamespace := gatewayPolicy.Namespace
		if policyNamespace == "" {
			policyNamespace = "default"
		}
		targetNS := targetGateway.Namespace
		if targetNS == "" {
			targetNS = policyNamespace
		}
		return []string{types.NamespacedName{Name: targetGateway.Name, Namespace: targetNS}.String()}
	}

	return []string{}
}
```

Reply:
Truly after looking at this I think I was just trying to avoid having to build out a namespacedname object, since I don't think there ever actually is a reason for it to use the policies namespace here.

Reply:
I'll just build it out so things are clearer

Reply:
>  I don't think there ever actually is a reason for it to use the policies namespace here.
I think we would use the policies namespace if it's left out of the target ref since the usual behavior there is to assume it's the same as the parent resource

Reply:
I see. Fair enough, I'll add that in.